# Processed E-commerce Dataset

## Introduction

Here, we will create a processed e-commerce dataset by combining information from Orders, Customers, and Products datasets.

We will use Pandas to load, explore, merge, concatenate, transform, and process the datasets. We will also perform DateTime operations on the order date and create useful derived columns.

Finally, the processed dataset will be exported as a CSV file.

## Objectives

The main objectives of this assignment are:

- Load the Orders, Customers, and Products datasets using Pandas.
- Explore the structure and contents of the datasets.
- Combine related datasets using `merge()`.
- Demonstrate combining DataFrames using `concat()`.
- Create or transform useful columns using `apply()`.
- Convert the order date into DateTime format.
- Extract month, day, and day of the week from the order date.
- Organize the data into a clean and meaningful DataFrame.
- Export the final processed dataset as a CSV file.

##### Import Required Library

In [19]:
import pandas as pd
print("Pandas imported successfully!")

Pandas imported successfully!


##### Load the Datasets
*The e-commerce data is provided in three separate CSV files: Orders, Customers, and Products.*

In [40]:
orders = pd.read_csv("Orders.csv")
customers = pd.read_csv("Customers.csv")
products = pd.read_csv("Products.csv")

In [41]:
# Preview the datasets
print("Orders Dataset:")
display(orders.head())

print("Customers Dataset:")
display(customers.head())

print("Products Dataset:")
display(products.head())

Orders Dataset:


,Order_ID,Order_Date,Customer_ID,Product_ID,Quantity,Payment_Method,Order_Status
0,O0001,2026-02-19,C027,P019,2,Credit Card,Delivered
1,O0002,2026-01-25,C006,P003,2,Debit Card,Delivered
2,O0003,2026-02-26,C015,P004,1,Cash on Delivery,Delivered
3,O0004,2026-03-04,C024,P015,3,Net Banking,Delivered
4,O0005,2026-03-29,C025,P009,5,Credit Card,Delivered


Customers Dataset:


,Customer_ID,Customer_Name,City,Region,Membership_Type
0,C001,Aarav Sharma,Srinagar,North,Premium
1,C002,Zoya Khan,Delhi,North,Regular
2,C003,Rohan Mehta,Mumbai,West,Premium
3,C004,Ananya Singh,Jammu,North,Regular
4,C005,Kabir Ali,Lucknow,North,New


Products Dataset:


,Product_ID,Product_Name,Category,Unit_Price,Brand
0,P001,Wireless Headphones,Electronics,1499,SoundMax
1,P002,Mechanical Keyboard,Electronics,2499,KeyPro
2,P003,Wireless Mouse,Electronics,899,TechGear
3,P004,Smart Watch,Electronics,3299,FitTech
4,P005,Power Bank,Electronics,1199,VoltPlus


##### Explore the Datasets

In [42]:
# Check the shape of each dataset
print("Orders Shape:", orders.shape)
print("Customers Shape:", customers.shape)
print("Products Shape:", products.shape)

# Check the column names
print("\nOrders Columns:", orders.columns.tolist())
print("Customers Columns:", customers.columns.tolist())
print("Products Columns:", products.columns.tolist())

Orders Shape: (120, 7)
Customers Shape: (30, 5)
Products Shape: (20, 5)

Orders Columns: ['Order_ID', 'Order_Date', 'Customer_ID', 'Product_ID', 'Quantity', 'Payment_Method', 'Order_Status']
Customers Columns: ['Customer_ID', 'Customer_Name', 'City', 'Region', 'Membership_Type']
Products Columns: ['Product_ID', 'Product_Name', 'Category', 'Unit_Price', 'Brand']


In [43]:
# Display dataset information
print("Orders Dataset Information:")
orders.info()

print("\nCustomers Dataset Information:")
customers.info()

print("\nProducts Dataset Information:")
products.info()

Orders Dataset Information:
<class 'pandas.DataFrame'>
RangeIndex: 120 entries, 0 to 119
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype
---  ------          --------------  -----
 0   Order_ID        120 non-null    str  
 1   Order_Date      120 non-null    str  
 2   Customer_ID     120 non-null    str  
 3   Product_ID      120 non-null    str  
 4   Quantity        120 non-null    int64
 5   Payment_Method  120 non-null    str  
 6   Order_Status    120 non-null    str  
dtypes: int64(1), str(6)
memory usage: 6.7 KB

Customers Dataset Information:
<class 'pandas.DataFrame'>
RangeIndex: 30 entries, 0 to 29
Data columns (total 5 columns):
 #   Column           Non-Null Count  Dtype
---  ------           --------------  -----
 0   Customer_ID      30 non-null     str  
 1   Customer_Name    30 non-null     str  
 2   City             30 non-null     str  
 3   Region           30 non-null     str  
 4   Membership_Type  30 non-null     str  
dtypes: str(5)


In [44]:
# Checking for Missing Values

print("Missing values in Orders:")
print(orders.isnull().sum())

print("\nMissing values in Customers:")
print(customers.isnull().sum())

print("\nMissing values in Products:")
print(products.isnull().sum())

Missing values in Orders:
Order_ID          0
Order_Date        0
Customer_ID       0
Product_ID        0
Quantity          0
Payment_Method    0
Order_Status      0
dtype: int64

Missing values in Customers:
Customer_ID        0
Customer_Name      0
City               0
Region             0
Membership_Type    0
dtype: int64

Missing values in Products:
Product_ID      0
Product_Name    0
Category        0
Unit_Price      0
Brand           0
dtype: int64


##### Merge Orders and Customers

*The Orders and Customers datasets are related through the `Customer_ID` column. We will use a left join to combine the customer information with the order records.*

In [46]:
# Merge Orders with Customers using Customer_ID
orders_customers = pd.merge(
    orders,
    customers,
    on="Customer_ID",
    how="left"
)

# Preview the merged dataset
print("Shape:", orders_customers.shape)
print("Columns:", orders_customers.columns.tolist())

orders_customers.head()

Shape: (120, 11)
Columns: ['Order_ID', 'Order_Date', 'Customer_ID', 'Product_ID', 'Quantity', 'Payment_Method', 'Order_Status', 'Customer_Name', 'City', 'Region', 'Membership_Type']


,Order_ID,Order_Date,Customer_ID,Product_ID,Quantity,Payment_Method,Order_Status,Customer_Name,City,Region,Membership_Type
0,O0001,2026-02-19,C027,P019,2,Credit Card,Delivered,Harsh Vardhan,Noida,North,Premium
1,O0002,2026-01-25,C006,P003,2,Debit Card,Delivered,Ishita Gupta,Bengaluru,South,Premium
2,O0003,2026-02-26,C015,P004,1,Cash on Delivery,Delivered,Karan Joshi,Chandigarh,North,Regular
3,O0004,2026-03-04,C024,P015,3,Net Banking,Delivered,Maryam Khan,Hyderabad,South,Regular
4,O0005,2026-03-29,C025,P009,5,Credit Card,Delivered,Reyansh Jain,Kolkata,East,New


##### Merge with Products

*The merged dataset is now combined with the Products dataset using `Product_ID` to add product-related information.*

In [47]:
# Merge the dataset with Products using Product_ID
processed_data = pd.merge(
    orders_customers,
    products,
    on="Product_ID",
    how="left"
)

# Preview the processed dataset
print("Shape:", processed_data.shape)
print("Columns:", processed_data.columns.tolist())

processed_data.head()

Shape: (120, 15)
Columns: ['Order_ID', 'Order_Date', 'Customer_ID', 'Product_ID', 'Quantity', 'Payment_Method', 'Order_Status', 'Customer_Name', 'City', 'Region', 'Membership_Type', 'Product_Name', 'Category', 'Unit_Price', 'Brand']


,Order_ID,Order_Date,Customer_ID,Product_ID,Quantity,Payment_Method,Order_Status,Customer_Name,City,Region,Membership_Type,Product_Name,Category,Unit_Price,Brand
0,O0001,2026-02-19,C027,P019,2,Credit Card,Delivered,Harsh Vardhan,Noida,North,Premium,Cricket Bat,Sports,2499,BatPro
1,O0002,2026-01-25,C006,P003,2,Debit Card,Delivered,Ishita Gupta,Bengaluru,South,Premium,Wireless Mouse,Electronics,899,TechGear
2,O0003,2026-02-26,C015,P004,1,Cash on Delivery,Delivered,Karan Joshi,Chandigarh,North,Regular,Smart Watch,Electronics,3299,FitTech
3,O0004,2026-03-04,C024,P015,3,Net Banking,Delivered,Maryam Khan,Hyderabad,South,Regular,Machine Learning Basics,Books,999,AIPress
4,O0005,2026-03-29,C025,P009,5,Credit Card,Delivered,Reyansh Jain,Kolkata,East,New,Coffee Maker,Home & Kitchen,3499,HomeBrew


##### Concatenate DataFrames

*The `concat()` function is used to combine DataFrames along a particular axis.*

*Here, the Orders dataset is divided into two parts and then concatenated vertically to demonstrate the use of `concat()`.*

In [48]:
# Split the Orders DataFrame into two parts
orders_part1 = orders.iloc[:60]
orders_part2 = orders.iloc[60:]

# Concatenate the two parts
orders_combined = pd.concat(
    [orders_part1, orders_part2],
    axis=0,
    ignore_index=True
)

# Preview the concatenated DataFrame
orders_combined.head()

,Order_ID,Order_Date,Customer_ID,Product_ID,Quantity,Payment_Method,Order_Status
0,O0001,2026-02-19,C027,P019,2,Credit Card,Delivered
1,O0002,2026-01-25,C006,P003,2,Debit Card,Delivered
2,O0003,2026-02-26,C015,P004,1,Cash on Delivery,Delivered
3,O0004,2026-03-04,C024,P015,3,Net Banking,Delivered
4,O0005,2026-03-29,C025,P009,5,Credit Card,Delivered


In [31]:
# Check the shape after concatenation
orders_combined.shape

(120, 7)

##### Create a Derived Column using `apply()`

*The `Quantity` and `Unit_Price` columns are used to calculate the total value of each order.*

*The `apply()` function is used to create a new `Order_Value` column.*

In [49]:
# Calculate the total value of each order
processed_data["Order_Value"] = processed_data.apply(
    lambda row: row["Quantity"] * row["Unit_Price"],
    axis=1
)

# Preview the updated dataset
processed_data.head()

,Order_ID,Order_Date,Customer_ID,Product_ID,Quantity,Payment_Method,Order_Status,Customer_Name,City,Region,Membership_Type,Product_Name,Category,Unit_Price,Brand,Order_Value
0,O0001,2026-02-19,C027,P019,2,Credit Card,Delivered,Harsh Vardhan,Noida,North,Premium,Cricket Bat,Sports,2499,BatPro,4998
1,O0002,2026-01-25,C006,P003,2,Debit Card,Delivered,Ishita Gupta,Bengaluru,South,Premium,Wireless Mouse,Electronics,899,TechGear,1798
2,O0003,2026-02-26,C015,P004,1,Cash on Delivery,Delivered,Karan Joshi,Chandigarh,North,Regular,Smart Watch,Electronics,3299,FitTech,3299
3,O0004,2026-03-04,C024,P015,3,Net Banking,Delivered,Maryam Khan,Hyderabad,South,Regular,Machine Learning Basics,Books,999,AIPress,2997
4,O0005,2026-03-29,C025,P009,5,Credit Card,Delivered,Reyansh Jain,Kolkata,East,New,Coffee Maker,Home & Kitchen,3499,HomeBrew,17495


##### DateTime Operations

*The `Order_Date` column is converted into DateTime format to extract useful information such as month, day, and day of the week.*

In [50]:
# Convert Order_Date to DateTime format
processed_data["Order_Date"] = pd.to_datetime(processed_data["Order_Date"])

# Extract date-related information
processed_data["Order_Month"] = processed_data["Order_Date"].dt.month
processed_data["Order_Day"] = processed_data["Order_Date"].dt.day
processed_data["Order_DayOfWeek"] = processed_data["Order_Date"].dt.day_name()

processed_data.head()

,Order_ID,Order_Date,Customer_ID,Product_ID,Quantity,Payment_Method,Order_Status,Customer_Name,City,Region,Membership_Type,Product_Name,Category,Unit_Price,Brand,Order_Value,Order_Month,Order_Day,Order_DayOfWeek
0,O0001,2026-02-19,C027,P019,2,Credit Card,Delivered,Harsh Vardhan,Noida,North,Premium,Cricket Bat,Sports,2499,BatPro,4998,2,19,Thursday
1,O0002,2026-01-25,C006,P003,2,Debit Card,Delivered,Ishita Gupta,Bengaluru,South,Premium,Wireless Mouse,Electronics,899,TechGear,1798,1,25,Sunday
2,O0003,2026-02-26,C015,P004,1,Cash on Delivery,Delivered,Karan Joshi,Chandigarh,North,Regular,Smart Watch,Electronics,3299,FitTech,3299,2,26,Thursday
3,O0004,2026-03-04,C024,P015,3,Net Banking,Delivered,Maryam Khan,Hyderabad,South,Regular,Machine Learning Basics,Books,999,AIPress,2997,3,4,Wednesday
4,O0005,2026-03-29,C025,P009,5,Credit Card,Delivered,Reyansh Jain,Kolkata,East,New,Coffee Maker,Home & Kitchen,3499,HomeBrew,17495,3,29,Sunday


##### Demonstrate `concat()`

*The `concat()` function is used to combine DataFrames with the same structure.
A small sample of the processed dataset is split into two parts and then concatenated back together.*

In [51]:
# Split the processed dataset into two parts
part1 = processed_data.iloc[:5]
part2 = processed_data.iloc[5:10]

# Combine the two parts using concat()
concat_demo = pd.concat([part1, part2], ignore_index=True)

# Display the concatenated DataFrame
concat_demo

,Order_ID,Order_Date,Customer_ID,Product_ID,Quantity,Payment_Method,Order_Status,Customer_Name,City,Region,Membership_Type,Product_Name,Category,Unit_Price,Brand,Order_Value,Order_Month,Order_Day,Order_DayOfWeek
0,O0001,2026-02-19,C027,P019,2,Credit Card,Delivered,Harsh Vardhan,Noida,North,Premium,Cricket Bat,Sports,2499,BatPro,4998,2,19,Thursday
1,O0002,2026-01-25,C006,P003,2,Debit Card,Delivered,Ishita Gupta,Bengaluru,South,Premium,Wireless Mouse,Electronics,899,TechGear,1798,1,25,Sunday
2,O0003,2026-02-26,C015,P004,1,Cash on Delivery,Delivered,Karan Joshi,Chandigarh,North,Regular,Smart Watch,Electronics,3299,FitTech,3299,2,26,Thursday
3,O0004,2026-03-04,C024,P015,3,Net Banking,Delivered,Maryam Khan,Hyderabad,South,Regular,Machine Learning Basics,Books,999,AIPress,2997,3,4,Wednesday
4,O0005,2026-03-29,C025,P009,5,Credit Card,Delivered,Reyansh Jain,Kolkata,East,New,Coffee Maker,Home & Kitchen,3499,HomeBrew,17495,3,29,Sunday
5,O0006,2026-02-09,C003,P018,3,UPI,Delivered,Rohan Mehta,Mumbai,West,Premium,Football,Sports,799,SportZone,2397,2,9,Monday
6,O0007,2026-02-10,C011,P003,3,UPI,Delivered,Vivaan Kapoor,Jaipur,North,New,Wireless Mouse,Electronics,899,TechGear,2697,2,10,Tuesday
7,O0008,2026-03-27,C020,P005,4,Debit Card,Delivered,Priya Menon,Chennai,South,Premium,Power Bank,Electronics,1199,VoltPlus,4796,3,27,Friday
8,O0009,2026-03-13,C024,P009,2,UPI,Cancelled,Maryam Khan,Hyderabad,South,Regular,Coffee Maker,Home & Kitchen,3499,HomeBrew,6998,3,13,Friday
9,O0010,2026-03-05,C026,P005,2,Debit Card,Shipped,Fatima Noor,Srinagar,North,Regular,Power Bank,Electronics,1199,VoltPlus,2398,3,5,Thursday


##### Organize the Final Processed Dataset

In [52]:
# Arrange the columns in an order
processed_data = processed_data[
    [
        "Order_ID",
        "Order_Date",
        "Order_Month",
        "Order_Day",
        "Order_DayOfWeek",
        "Customer_ID",
        "Customer_Name",
        "City",
        "Region",
        "Membership_Type",
        "Product_ID",
        "Product_Name",
        "Category",
        "Brand",
        "Quantity",
        "Unit_Price",
        "Order_Value",
        "Payment_Method",
        "Order_Status"
    ]
]

# Verify the final processed dataset
print("Shape:", processed_data.shape)
print("Columns:", processed_data.columns.tolist())

processed_data.head()

Shape: (120, 19)
Columns: ['Order_ID', 'Order_Date', 'Order_Month', 'Order_Day', 'Order_DayOfWeek', 'Customer_ID', 'Customer_Name', 'City', 'Region', 'Membership_Type', 'Product_ID', 'Product_Name', 'Category', 'Brand', 'Quantity', 'Unit_Price', 'Order_Value', 'Payment_Method', 'Order_Status']


,Order_ID,Order_Date,Order_Month,Order_Day,Order_DayOfWeek,Customer_ID,Customer_Name,City,Region,Membership_Type,Product_ID,Product_Name,Category,Brand,Quantity,Unit_Price,Order_Value,Payment_Method,Order_Status
0,O0001,2026-02-19,2,19,Thursday,C027,Harsh Vardhan,Noida,North,Premium,P019,Cricket Bat,Sports,BatPro,2,2499,4998,Credit Card,Delivered
1,O0002,2026-01-25,1,25,Sunday,C006,Ishita Gupta,Bengaluru,South,Premium,P003,Wireless Mouse,Electronics,TechGear,2,899,1798,Debit Card,Delivered
2,O0003,2026-02-26,2,26,Thursday,C015,Karan Joshi,Chandigarh,North,Regular,P004,Smart Watch,Electronics,FitTech,1,3299,3299,Cash on Delivery,Delivered
3,O0004,2026-03-04,3,4,Wednesday,C024,Maryam Khan,Hyderabad,South,Regular,P015,Machine Learning Basics,Books,AIPress,3,999,2997,Net Banking,Delivered
4,O0005,2026-03-29,3,29,Sunday,C025,Reyansh Jain,Kolkata,East,New,P009,Coffee Maker,Home & Kitchen,HomeBrew,5,3499,17495,Credit Card,Delivered


##### Export CSV

In [53]:
# Export the processed dataset as a CSV file
processed_data.to_csv("Processed_Ecommerce_Dataset.csv", index=False)

print("Processed dataset exported successfully.")

Processed dataset exported successfully.


In [39]:
# Display the final processed dataset
processed_data.head()

,Order_ID,Order_Date,Order_Month,Order_Day,Order_DayOfWeek,Customer_ID,Customer_Name,City,Region,Membership_Type,Product_ID,Product_Name,Category,Brand,Quantity,Unit_Price,Order_Value,Payment_Method,Order_Status
0,O0001,2026-02-19,2,19,Thursday,C027,Harsh Vardhan,Noida,North,Premium,P019,Cricket Bat,Sports,BatPro,2,2499,4998,Credit Card,Delivered
1,O0002,2026-01-25,1,25,Sunday,C006,Ishita Gupta,Bengaluru,South,Premium,P003,Wireless Mouse,Electronics,TechGear,2,899,1798,Debit Card,Delivered
2,O0003,2026-02-26,2,26,Thursday,C015,Karan Joshi,Chandigarh,North,Regular,P004,Smart Watch,Electronics,FitTech,1,3299,3299,Cash on Delivery,Delivered
3,O0004,2026-03-04,3,4,Wednesday,C024,Maryam Khan,Hyderabad,South,Regular,P015,Machine Learning Basics,Books,AIPress,3,999,2997,Net Banking,Delivered
4,O0005,2026-03-29,3,29,Sunday,C025,Reyansh Jain,Kolkata,East,New,P009,Coffee Maker,Home & Kitchen,HomeBrew,5,3499,17495,Credit Card,Delivered


### Key Insights

- *The final dataset combines order, customer, and product information into a single meaningful DataFrame.*
- *The order date was converted into DateTime format, and the month, day, and day of the week were extracted for further analysis.*
- *The `Order_Value` was calculated using quantity and unit price, which helps in understanding the value of each order.*
- *The processed dataset contains 120 orders and 19 columns.*
- *The merged dataset provides useful information about customers, products, regions, membership types, payment methods, and order status in one place.*